mypy/subtypes.py

```diff
@@ -1382,9 +1375,17 @@ def check_argument(leftarg: Type, rightarg: Type, variance: int) -> bool:
                 nominal = True
                 for ta, ra, tvar in zip(left.args, right.args, right.type.defn.type_vars):
                     if isinstance(tvar, TypeVarType):
-                        nominal = nominal and check_argument(ta, ra, tvar.variance)
+                        variance = tvar.variance
+                        if variance == COVARIANT:
+                            nominal = nominal and self._is_proper_subtype(ta, ra)
+                        elif variance == CONTRAVARIANT:
+                            nominal = nominal and self._is_proper_subtype(ra, ta)
+                        else:
+                            nominal = nominal and mypy.sametypes.is_same_type(ta, ra)
                     else:
                         nominal = nominal and mypy.sametypes.is_same_type(ta, ra)
```

Comment:
```suggestion
                        variance = tvar.variance
                        if variance == COVARIANT:
                            nominal = self._is_proper_subtype(ta, ra)
                        elif variance == CONTRAVARIANT:
                            nominal = self._is_proper_subtype(ra, ta)
                        else:
                            nominal = mypy.sametypes.is_same_type(ta, ra)
                    else:
                        nominal = mypy.sametypes.is_same_type(ta, ra)
```
These are all redundant now